packages/agents/src/agents/Chameleon/ContextualizedChat.ts

```diff
@@ -0,0 +1,294 @@
+import {
+  Chat,
+  ChatLogType,
+  ChatMessage,
+  MessageChunker,
+  LocalVectorDB,
+  LocalCollection
+} from "@evo-ninja/agent-utils";
+import { v4 as uuid } from "uuid";
+
+type MsgPtr = number;
+
+interface ChunkMetadata {
+  msgPtr: MsgPtr;
+}
+
+type ChunkMetadataPtr = number;
+
+interface DocumentMetadata {
+  chunkMetadataPtr: ChunkMetadataPtr;
```

Comment:
I'd look into having this prop be passed around instead of it being a prop, as its side-effects make it harder to track its mutations. If we're passing a ptr around just to get it from here, we might as well simplify and just pass around the value

Also, probably it'd benefit from a name change, as I see it's used more as a "currentProcessedMessages" or something along those lines

Reply:
Updated naming to be clearer